src/app/pages/users/user-list/user-list.page.html

```diff
@@ -6,7 +6,7 @@
 
 <ion-content>
   <ion-list>
-    <ion-item *ngFor='let user of users'>
+    <ion-item *ngFor='let user of users' [routerLink]="['details/' + user._id]">
```

Comment:
Lasciamo che la route rifletta quella dell' API: `users/:id`

Reply:
Quindi invece di `users/:id`
diventerà: `[routerLink]="['students/' + user._id]"` oppure `[routerLink]="['teachers/' + user._id]"` in base allo `user.type`. Puoi crearti un condizionale nel file.ts che ti torna una path in base al type.